src/js/on-page-load.js

```diff
@@ -0,0 +1,57 @@
+import { getMovies } from "./get-movies.js";
+
+const addCardsToSwiper = (data, id)=> {
+    const swiperName = document.getElementById(id);
+    const slides = swiperName.querySelector(".swiper-wrapper")
+
+    for (const el of data) {
+        const swiperSlide = document.createElement("div");
+        swiperSlide.classList.add('swiper-slide');
+        const card = document.createElement("div");
+        card.classList.add("card-container");
+
+        if (el?.primaryImage) {
+            card.dataset.bgImage = el.primaryImage;
+            const observer = new IntersectionObserver(
+                (entries, observer) => {
+                    entries.forEach((entry) => {
+                        if (entry.isIntersecting) {
+                            const target = entry.target;
```

Comment:
Using IntersectionObserver for each card individually can be inefficient, especially if there are a lot of cards. It creates multiple observers, which could impact performance.